components/script/dom/validation.rs

```diff
@@ -1,5 +1,10 @@
 /* This Source Code Form is subject to the terms of the Mozilla Public
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+use dom::element::Element;
+use dom::validitystate::ValidityStatus;
 
-pub trait Validatable {}
+pub trait Validatable {
+    fn as_element_validatable(&self) -> &Element;
+    fn is_instance_validatable(&self) -> bool;
+}
```

Comment:
We will need a method added here that performs the appropriate validation checks, returns true if the element validates, and false otherwise.